src/openPr.js

```diff
@@ -73,11 +78,44 @@ module.exports = async function ({ context, inputs, packageVersion }) {
 
   logInfo(`New version ${newVersion}`)
 
+  // attach artifact
+  const artifactPath = inputs['artifact-path']
+
+  let artifact = {
+    isPresent: false,
+    url: null,
+    label: null,
+  }
+
+  if (artifactPath) {
+    logInfo('** Attaching artifact **')
+
+    const token = inputs['github-token']
+    const artifactFilename = inputs['artifact-filename']
+    const artifactLabel = inputs['artifact-label']
+    const { id: releaseId } = draftRelease
+    try {
+      ;({ artifact } = await attachArtifact(
+        artifactPath,
+        artifactFilename,
+        artifactLabel,
+        releaseId,
+        token
+      ))
+    } catch (err) {
+      logInfo(err.message)
+      core.setFailed(err.message)
+    }
+
+    logInfo('** Artifact attached! **')
+  }
```

Comment:
let's avoid mutable state. iow, let's avoid let artifact, then the weird ;({ ... }) = await syntax. also, let's extract this in a function please, this function is becoming too complex